### PR TITLE
data-fetching-astro-glob

### PR DESCRIPTION
**Pull request by sarah11918**
- New or updated content

A quick example of `Astro.glob()` on the data fetching page until we have a proper "Here's how to Content with Astro" full story. This is better exposure for an example of creating an index of blog posts, which currently only exists in Reference and is not as accessible.

Now, this page details:
- Fetching data from local files to use in a component (`Astro.glob`)
in addition to what was previously there:
- fetch from remote APIs, incl GraphQL, in both Astro and UI Framework components
- fetch from a headless CMS

### CODE REVIEW

File: src/pages/en/guides/data-fetching.md [hunk 1]
```diff
@@ -5,7 +5,24 @@ description: Learn how to fetch remote data with Astro using the fetch API.
 i18nReady: true
 ---
 
-`.astro` files can fetch remote data at build time to help generate your pages.
+`.astro` files can fetch data from your local files, or remote data at build time to help generate your pages.
```
Comment (Yan-Thomas) [inline src/pages/en/guides/data-fetching.md:8]: This sentence feels a bit wordy to me, what do you think of rephrasing it this way? I omitted local *files* but since the example mentions that we are fetching from a local file I think readers aren't going to be confused.
```suggestion
`.astro` files can fetch local and remote data at build time to help generate your pages.
```

**Review by Yan-Thomas — COMMENTED** [review]
Thanks for this @sarah11918, nice work! :raised_hands: I added a suggestion but LGTM!

File: src/pages/en/guides/data-fetching.md [hunk 2]
```diff
@@ -5,7 +5,24 @@ description: Learn how to fetch remote data with Astro using the fetch API.
...
+Get data from one or many local files in your project using Astro's top-level await and [`Astro.glob()`](/en/reference/api-reference/#astroglob). This function returns an array (even if you only query one file!) and gives your `.astro` file access to exported properties like Markdown/MDX frontmatter and more!
+
+```astro "Astro.glob('../pages/post/*.md')"
+---
+// src/components/blog-archive.astro
+const posts = await Astro.glob('../pages/posts/*.md');
+---
+  <ul>
+    {posts.map((post) => (
+      <li><a href={post.frontmatter.url}>{post.frontmatter.title}</li></p>
```
Comment (sarah11918) [inline src/pages/en/guides/data-fetching.md:21]:
```suggestion
      <li><a href={post.frontmatter.url}>{post.frontmatter.title}</a></li>
```

**Comment by delucis** [comment]
Just a note to say @sarah11918 & I did discuss this and concluded that this didn’t really belong on data “fetching” because that is usually understood to mean “remote” data. We can keep this PR around for reference if it’s useful for a future content-focused guide, but otherwise this can be closed!
